src/engine/renderer/glsl_source/reliefMapping_fp.glsl

```diff
@@ -134,7 +134,13 @@ vec2 ParallaxTexOffset(vec2 rayStartTexCoords, vec3 viewDir, mat3 tangentToWorld
 	{
 		currentSize *= 0.5;
 
-		float heightMapDepth = topDepth - texture2D(u_NormalMap, rayStartTexCoords + displacement * currentDepth).a;
+#if defined(USE_HEIGHTMAP_IN_NORMALMAP)
+		float depth = texture2D(u_HeightMap, rayStartTexCoords + displacement * currentDepth).a;
```

Comment:
This should say `u_normalMap`?

Reply:
The code is correct because when an heightmap is in normalmap, normalmap is bound as heightmap and glsl code reads heightmap alpha. Well, we may avoid this bind by putting `uniform sampler2D	u_HeightMap;` in a `USE_HEIGHTMAP_IN_NORMALMAP` ifdef as you said. That may be better performance wise.

I wrote this code as if I was able to do swizzle masks in the future, so I would just bind the normalmap as heightmap and swizzle alpha to red, or bind heightmap as heightmap and swizzle nothing, and read heightmap red in all case in glsl. 

By doing swizzle masks, we would also be able to avoid the `USE_HEIGHTMAP_IN_NORMALMAP` macro entirely by setting to 0 the alpha channel when feeding normalmap as normalmap to glsl code.

I'm a bit concerned by the way an additional bound would affect a bit the performance.  On the other hand, implementing swizzle mask would reduce the number of shaders to compile. But since shader compilation is only done once at map loading, maybe I can rewrite that code to avoid that extra bind when heightmap is embedded in normalmap. This is about deciding we would never get rid of the `USE_HEIGHTMAP_IN_NORMALMAP` macro that duplicates shaders.

Reply:
I pushed a commit that does many tests to avoid the extra bind. The code may be faster but the code is far more convoluted. Do you think this is better?

Reply:
I'm experimenting on a branch about generating normalmap from heightmap when only heightmap is provided (Xonotic has some textures like that). That other patch I'm working on integrates well with the more-convoluted-code-less-texture-bind implementation so I'm now fully ok for merging that.

Reply:
2nd commit seems good to me